src/main/java/com/android/volley/toolbox/DiskBasedAsyncCache.java

```diff
@@ -411,4 +405,67 @@ private void deleteFileAndInvokeCallback(
         mTotalSize = DiskBasedCacheUtility.removeEntry(key, mTotalSize, mEntries);
         callback.onGetComplete(null);
     }
+
+    /**
+     * Builder is used to build an instance of {@link DiskBasedAsyncCache} from values configured by
+     * the setters.
+     */
+    public static class Builder {
+        @Nullable private FileSupplier rootDirectorySupplier;
+        @Nullable private File rootDirectory;
+        private int maxCacheSizeInBytes;
+
+        public Builder() {
+            rootDirectorySupplier = null;
+            rootDirectory = null;
+            maxCacheSizeInBytes = -1;
+        }
+
+        /**
+         * Sets the root directory of the cache. Must be called if {@link
+         * Builder#setRootDirectorySupplier(FileSupplier)} is not.
+         */
+        public Builder setRootDirectory(File file) {
+            this.rootDirectory = file;
+            return this;
+        }
+
+        /**
+         * Sets the root directory supplier of the cache. Must be called if {@link
+         * Builder#setRootDirectory(File)} is not.
+         */
+        public Builder setRootDirectorySupplier(FileSupplier fileSupplier) {
+            this.rootDirectorySupplier = fileSupplier;
+            return this;
+        }
+
+        /**
+         * Sets the max size of the cache in bytes. Will default to {@link
+         * DiskBasedCacheUtility#DEFAULT_DISK_USAGE_BYTES} if not called.
+         */
+        public Builder setMaxCacheSizeInBytes(int maxCacheSizeInBytes) {
+            this.maxCacheSizeInBytes = maxCacheSizeInBytes;
+            return this;
+        }
+
+        /** Builds a DiskBasedAsyncCache from the provided parameters. */
+        public DiskBasedAsyncCache build() {
+            if (rootDirectory == null && rootDirectorySupplier == null) {
+                throw new IllegalArgumentException("Must set either file or supplier");
+            }
+            if (maxCacheSizeInBytes == -1) {
+                maxCacheSizeInBytes = DiskBasedCacheUtility.DEFAULT_DISK_USAGE_BYTES;
+            }
```

Comment:
What if the default value was `DEFAULT_DISK_USAGE_BYTES` instead of `-1`?

Reply:
Oh yep, that saves us from writing this code block in build().